tutorials/pubsub-spring-dedup-messages/index.md

```diff
@@ -0,0 +1,263 @@
+
+In [DataEntryController.java](pubsub-spring/src/main/java/com/google/example/DataEntryController.java):
+
+[embedmd]:# (pubsub-spring/src/main/java/com/google/example/DataEntryController.java java /  \/\/ Headers/ /;/)
+```java
+  // Headers become Pub/Sub message attributes.
+    Message<String> message = MessageBuilder
+      .withPayload(data)
+      .setHeader("key", key)
+      .build();
+```
+
+### Run the application
+To start your application, navigate to `pubsub-spring/` and run: 
+
+```shell script
+mvn spring-boot:run
+```
+
+Check that your app has started successfully by pointing your browser to `localhost:8080`. You should be able to send messages using the form there. At this point, Spring will have automatically created the Cloud Pub/Sub topic that you have specified in `application.properties` to publish your message to. You can view Publish Message Request Count and Publish Message Operation Count in the topic details in [Cloud Console for Pub/Sub Topic] to verify that publishing to Cloud Pub/Sub is successful.  
+
+## Start a Cloud Dataflow Job to Deduplicate Pub/Sub Messages
+
+As a middle process that takes data from a Cloud Pub/Sub topic and publishes processed data to another Cloud Pub/Sub topic, Cloud Dataflow achieves exactly once stream processing by asking the input stream for an `idAttribute`. The `idAttribute` can be a Pub/Sub message attribute. Cloud Dataflow will avoid processing messages that have the same values in this `idAttribute` repeatedly.
+
+In [DedupPubSub.java](pubsubio-dedup/src/main/java/com/google/example/DedupPubSub.java):
+
+[embedmd]:# (pubsubio-dedup/src/main/java/com/google/example/DedupPubSub.java java /  pipeline\n.*1\)./ /;/)
+```java
+  pipeline
+        // 1) Read string messages with attributes from a Pub/Sub topic.
+        .apply(
+            "Read from PubSub",
+            PubsubIO.readStrings()
+                .fromTopic(options.getInputTopic())
+                .withIdAttribute(options.getIdAttribute()))
+        // 2) Write string messages to another Pub/Sub topic.
+        .apply("Write to PubSub", PubsubIO.writeStrings().to(options.getOutputTopic()));
+```
+
+### Run the Dataflow job
+
+To start the Dataflow job,  navigate to `pubsubio-dedup/` and run: 
+```shell script
+  mvn compile exec:java \
+   -Dexec.mainClass=com.google.example.DedupPubSub \
+   -Dexec.cleanupDaemonThreads=false \
+   -Dexec.args="\
+     --project=$PROJECT_NAME \
+     --inputTopic=projects/$PROJECT_NAME/topics/topicToDataflow \
+     --outputTopic=projects/$PROJECT_NAME/topics/topicFromDataflow \
+     --idAttribute=key \
+     --runner=DataflowRunner"
+```
+You can check the job's progress in the [Cloud Console for Dataflow]. Wait a few minutes for the job status to become "running". Then issue `Ctrl+C` to stop the program locally. Stopping the program locally does not affect the Dataflow job running in the Cloud.
```

Comment:
How come "running" is quotes?

Reply:
I was quoting this screenshot.

![Screen Shot 2020-02-13 at 5 38 24 PM](https://user-images.githubusercontent.com/9518298/74577880-6a6c2a00-4f46-11ea-8561-ffece3aad490.png)

I think I will add this screenshot. Wdyt?

Reply:
The screenshot could help. In either case, I think **Status** and **Running** should be in title case + bold since they refer to elements in the UI. You may need to rephrase the sentence a bit.

Reply:
Sounds good! Will add a screenshot and bold + upper case text here.